hudi-common/src/main/java/org/apache/hudi/metadata/HoodieTableMetadataUtil.java

```diff
@@ -1683,11 +1688,11 @@ private static Comparable<?> coerceToComparable(Schema schema, Object val) {
   }
 
   private static boolean isColumnTypeSupported(Schema schema, Option<HoodieRecordType> recordType) {
-    // if record type is set and if its AVRO, MAP is unsupported.
-    if (recordType.isPresent() && recordType.get() == HoodieRecordType.AVRO) {
+    // if record type is not set or if its AVRO, MAP is unsupported.
+    if (!recordType.isPresent() || recordType.get() == HoodieRecordType.AVRO) {
```

Comment:
let's add a test for:
a) nested field
b) field configured to index in metadata config but its data type is not supported. In this case, I expect to either error out and have no update in index definition, or only update supported fields in the ndex definition.